### ISSUE TITLE
unit tests from MarkDuplicates

### ISSUE THREAD

**Issue by delocalizer** [delocalizer/streammd#25]
https://twitter.com/nilshomer/status/1582603486629199873?t=dy0HNME-WofbtnTNdzzaWA&s=19

**Comment by delocalizer** [comment]
scope of this:

```bash
#cd picard/src/test/java/picard/sam/markduplicates
#grep -sl '\.runTest()' *|xargs grep -c '\.runTest()'
AbstractMarkDuplicatesCommandLineProgramTest.java:54
AsIsMarkDuplicatesTest.java:2
MarkDuplicateWithMissingBarcodeTest.java:1
MarkDuplicatesForFlowHelperTest.java:1
MarkDuplicatesSetSizeHistogramTest.java:4
MarkDuplicatesTagRepresentativeReadIndexTest.java:2
MarkDuplicatesTest.java:7
MarkDuplicatesTestQueryNameSorted.java:1
MarkDuplicatesWithMateCigarTest.java:6
UmiAwareMarkDuplicatesWithMateCigarTest.java:6
```
by inspection, given the current functionality of `streammd` the tests that it's meaningful to replicate are in
```
AbstractMarkDuplicatesCommandLineProgramTest.java
AsIsMarkDuplicatesTest.java
MarkDuplicatesTest.java
```

**Comment by delocalizer** [comment]
many thanks to @nh13 :pray:

**Comment by delocalizer** [comment]
these tests are available in [4.1.4](https://github.com/delocalizer/streammd/releases/tag/4.1.4)
